app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  # Associations
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # Attributes
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  # Methods
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
```

Comment:
[OPTIONAL] It is a good practice to use rails callbacks like `after_save` in scenarios like this to call the custom method after creation.

Reply:
Addressed